pkg/clusterapi/system.go

```diff
@@ -195,6 +195,7 @@ func (c *system) Run(ctx context.Context) error {
 					"-webhook-cert-dir={{.WebhookCertDir}}",
 					"-crd-pattern=",
 					"-crd-management=none",
+					"--feature-gates=MachinePool=false",
```

Comment:
This is set in the wrong controller. You want it in Azure, not AzureASO

Reply:
Yup. Just noticed the CI failure.